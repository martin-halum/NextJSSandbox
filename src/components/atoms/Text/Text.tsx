import React from 'react';
import { Typography } from '@mui/material';

import type { PropsType } from './types';

const Text = (props: PropsType) => {
    const {
        align,
        children,
        classes,
        component,
        gutterBottom,
        noWrap,
        paragraph,
        styles,
        variant,
        variantMapping
    } = props;

    return (
        <Typography
            align={align}
            classes={classes}
            component={component}
            gutterBottom={gutterBottom}
            noWrap={noWrap}
            paragraph={paragraph}
            sx={styles}
            variant={variant}
            variantMapping={variantMapping}
        >
            {children}
        </Typography>
    );
};


export default Text;
