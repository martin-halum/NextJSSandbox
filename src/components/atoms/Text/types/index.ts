import { SxProps, Theme } from "@mui/material";
import { ElementType, ReactNode } from "react";

export type PropsType = {
    align?: 'center' | 'inherit' | 'justify' | 'left' | 'right';
    children?: ReactNode;
    classes?: Object;
    component?: ElementType;
    gutterBottom?: boolean;
    noWrap?: boolean;
    paragraph?: boolean;
    styles?: SxProps<Theme>;
    variant?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "subtitle1" | "subtitle2" | "body1" | "body2" | "caption" | "button" | "overline" | "inherit";
    variantMapping?: Object;
};
