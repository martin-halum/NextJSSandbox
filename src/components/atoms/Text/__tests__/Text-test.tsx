import React from 'react';
import renderer from 'react-test-renderer';

import Text from '../Text';

describe('Text Test', () => {
  test('renders the Text String', () => {
    const tree = renderer
      .create(
        <Text>Test</Text>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})