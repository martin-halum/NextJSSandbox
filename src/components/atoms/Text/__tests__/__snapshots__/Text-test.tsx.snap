// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Test renders the Text String 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-ahj2mt-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;}",
      }
    }
    data-emotion="css ahj2mt-MuiTypography-root"
  />,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Test
  </p>,
]
`;
